Compilation failure under wine

On some machines, when building under wine from linux, CL emits some weird errors:

```
[22/23] cl.exe /nologo /showIncludes /MT /EHsc /G5 /Gs /GS /DNDEBUG /Zi /I build/autogenerated /I src /I ...ty/Detours/src /Od /Oi /Ob1 /Op /c src/Supervisor.cpp /Fdbuild/Supervisor.obj.pdb /Fobuild/Supervisor.obj
FAILED: cl.exe /nologo /showIncludes /MT /EHsc /G5 /Gs /GS /DNDEBUG /Zi /I build/autogenerated /I src /I src/pbg3 /I 3rdparty/munit /I 3rdparty/Detours/src /Od /Oi /Ob1 /Op /c src/Supervisor.cpp /Fdbuild/Supervisor.obj.pdb /Fobuild/Supervisor.obj
th06\src\ZunResult.hpp(4) : error C2011: 'ZunResult' : 'enum' type redefinition
        th06\src\ZunResult.hpp(4) : see declaration of 'ZunResult'
[22/23] cl.exe /nologo /showIncludes /MT /EHsc /G5 /Gs /GS /DNDEBUG /Zi /I build/autogenerated /I src /I ...ours/src /Od /Oi /Ob1 /Op /c src/AsciiManager.cpp /Fdbuild/AsciiManager.obj.pdb /Fobuild/AsciiManager.obj
FAILED: cl.exe /nologo /showIncludes /MT /EHsc /G5 /Gs /GS /DNDEBUG /Zi /I build/autogenerated /I src /I src/pbg3 /I 3rdparty/munit /I 3rdparty/Detours/src /Od /Oi /Ob1 /Op /c src/AsciiManager.cpp /Fdbuild/AsciiManager.obj.pdb /Fobuild/AsciiManager.obj
th06\src\ZunResult.hpp(4) : error C2011: 'ZunResult' : 'enum' type redefinition
        th06\src\ZunResult.hpp(4) : see declaration of 'ZunResult'
th06\src\Chain.hpp(9) : error C2011: 'ChainElemFlag' : 'enum' type redefinition
        th06\src\Chain.hpp(9) : see declaration of 'ChainElemFlag'
th06\src\Chain.hpp(15) : error C2011: 'ChainCallbackResult' : 'enum' type redefinition
        th06\src\Chain.hpp(15) : see declaration of 'ChainCallbackResult'
th06\src\Chain.hpp(31) : error C2011: 'ChainElem' : 'class' type redefinition
        th06\src\Chain.hpp(31) : see declaration of 'ChainElem'
th06\src\Chain.hpp(49) : error C2011: 'Chain' : 'class' type redefinition
        th06\src\Chain.hpp(49) : see declaration of 'Chain'
```

This seems to go away when we build with ninja `-j1`.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -24,7 +24,7 @@ jobs:
       id: install_prefix
       run: python scripts/create_devenv.py scripts/dls scripts/prefix
     - name: Build binary
-      run: python scripts/build.py
+      run: python scripts/build.py -j0
     - uses: actions/upload-artifact@v3
       with:
         name: th06e
@@ -57,7 +57,7 @@ jobs:
       id: install_prefix
       run: python scripts/create_devenv.py scripts/dls scripts/prefix
     - name: Build binary
-      run: python scripts/build.py --build-type diffbuild
+      run: python scripts/build.py -j0 --build-type diffbuild
     - uses: actions/upload-artifact@v3
       with:
         name: th06e-diff
@@ -92,7 +92,7 @@ jobs:
       id: install_prefix
       run: python scripts/create_devenv.py scripts/dls scripts/prefix
     - name: Build binary
-      run: python scripts/build.py --build-type dllbuild
+      run: python scripts/build.py -j0 --build-type dllbuild
     - uses: actions/upload-artifact@v3
       with:
         name: th06e-dll

--- a/scripts/build.py
+++ b/scripts/build.py
@@ -7,12 +7,15 @@ from winhelpers import run_windows_program
 SCRIPTS_DIR = Path(__file__).parent
 
 
-def build(build_type, verbose=False):
+def build(build_type, verbose=False, jobs=1):
     configure(build_type)
 
     ninja_args = []
     if verbose:
         ninja_args += ["-v"]
+
+    if jobs != 0:
+        ninja_args += ["-j" + str(jobs)]
 
     if build_type == BuildType.TESTS:
         ninja_args += ["build/th06e-tests.exe"]
@@ -37,6 +40,14 @@ def main():
         choices=["normal", "diffbuild", "tests", "dllbuild"],
         default="normal",
     )
+    parser.add_argument(
+        "-j",
+        "--jobs",
+        type=int,
+        default=1,
+        help="""Number of jobs to run in parallel. Set to 0 to run one job per CPU core. Defaults to 1.
+        Note that parallel builds may not work when running through wine. See https://github.com/happyhavoc/th06/issues/79 for more information.""",
+    )
     parser.add_argument("--verbose", action="store_true")
     args = parser.parse_args()
 
@@ -50,7 +61,7 @@ def main():
     elif args.build_type == "dllbuild":
         build_type = BuildType.DLLBUILD
 
-    build(build_type, args.verbose)
+    build(build_type, args.verbose, args.jobs)
 
 
 if __name__ == "__main__":

